include/cpptoml.h

```diff
@@ -307,9 +323,27 @@ class value : public base
 
   private:
     T data_;
+
+    /**
+     * Constructs a value from the given data.
+     */
+    value(const T& val) : data_(val)
+    {
+    }
+
+    value(const value& val) = delete;
+    value& operator=(const value& val) = delete;
 };
 
 template <class T>
+inline std::shared_ptr<typename value_traits<T>::type> make_value(T&& val)
+{
+    typedef std::shared_ptr<typename value_traits<T>::type> pointer_type;
+    return pointer_type(
+        new typename pointer_type::element_type(std::forward<T>(val)));
+}
```

Comment:
We should use `make_shared()` here since it enables an optimization (standard library can allocate the object and the control block all in one go). It's a little tricky, though, but (I think) I've got an example working:

```cpp
template <class T>
std::shared_ptr<typename value_traits<T>::type> make_value(T&& val) {
    using value_type = typename value_traits<T>::type;
    struct make_shared_enabler : public value_type
    {
        make_shared_enabler(T&& val) : value_type(std::forward<T>(val))
        {
            // nothing
        }
    };
    return std::make_shared<make_shared_enabler>(std::forward<T>(val));
}
```

This uses a local class that *does* have a public ctor that's within friend level scope to allow the standard library to still be able to invoke it from outside the friend scope. Regular users won't have access to this "key", since it's a local class.